### ISSUE TITLE
STS4 should include LiveReload support on internal browser

### ISSUE THREAD

**Issue by chexmo** [spring-projects/sts4#526]
Not a dramatic issue, but i believe that if Spring Boot provides that functionality bundled, Spring tools must to.

**Comment by BoykoAlex** [comment]
Isn't it already provided by adding DevTools to your project via STS? Right-click on the project **Spring** -> **Add DevTools**...

**Comment by martinlippert** [comment]
I think there are two different things discussed here: one is the "automatic restart" of devtools (https://docs.spring.io/spring-boot/docs/current/reference/html/using-spring-boot.html#using-boot-devtools-restart), for which you can add devtools to the project and you are done. I think this is what @BoykoAlex has in mind.

The other feature is the live reload in the browser (https://docs.spring.io/spring-boot/docs/current/reference/html/using-spring-boot.html#using-boot-devtools-restart). This requires a live reload browser extension to work, which is available for Chrome, Firefox, and Safari.

If I understand the feature request correctly, it is asking for such a live reload extension for the embedded browser in the IDE - wich could be indeed a nice enhancement for the embedded browser experience.

I am not sure though how many users are really using the embedded browser for testing their apps, since it is quite limited. I guess that most users configured an external browser to be used instead, which you can easily do from the preferences (`General -> Web Browser`). Once configured, all the browser interactions (like clicking on the boot dashboard app node to jump to the browser) would use the external "real" browser app, which I think most users prefer.

@chexmo What do you think? Do you prefer the embedded browser experience over using the external one?

**Comment by chexmo** [comment]
Yeah, i'm talking about the second one. You add DevTools when creating the project and then the internal browser doesn't suppport it.
I don't ignore the fact that most of the web developers use their favorite browser instead of the embedded one, and I also know that the main use of spring tends more to RESTful services than MVC services... (So it should be more handy a Postman-like feature than the browser one)
So... What I say is.... if the embedded browser was better, maybe it would be more used.
However I understand what you're looking at: Adding features take time and effort, there is no point in waste resources developing something the community won't use lately.
That's why I post this, maybe i'm not alone with this. Who knows?

**Comment by martinlippert** [comment]
> That's why I post this, maybe i'm not alone with this. Who knows?

@chexmo Yes, absolutely. We can use this issue here for the community to vote it up if more people are interested. So please provide your thumbs-up on this in case you would prefer to use the embedded web browser and spread the word about this.

**Comment by manueljordan** [comment]
>I am not sure though how many users are really using the embedded browser for testing their apps, since it is quite limited

Agree, but assume it would be enhanced in the future by Eclipse, with some internal dev tools, such as Chrome with the F12.

Other situation is that seems that Live Reload is "frozen", I mean, I don't see any updates, I want assume either it is final or it does not require more fixes or something new.

Some reason to use the internal web browser is when the PC would be old and the memory ram is running out of capacity. So instead to open a new browser (Chrome/FF) where in mac consumes more resources than in Windows, specially about the "Helper Windows" ... , the internal Web Browser within the IDE fits well, but is limited until now.

A problem with live reload is that only offers so far right officially the extension for Chrome, the link to FF from its homepage is not valid anymore, and doing a search appears many options, not sure what could be the official. Not sure the case for Safari.

>So... What I say is.... if the embedded browser was better, maybe it would be more used.

I am assuming that a bug/issue ticket was already created by some web developer on Eclipse, right?

Or perhaps is there a plugin around there?

**Comment by martinlippert** [comment]
Closing this one, no plans to work on this in the near future.
